src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -30,10 +30,10 @@ internal partial class CSharpSimplificationService()
     // 2. Extension method reducer may insert parentheses.  So run it before the parentheses remover.
     private static readonly ImmutableArray<AbstractReducer> s_reducers =
         [
+            new CSharpCastReducer(),
             new CSharpVarReducer(),
             new CSharpNameReducer(),
             new CSharpNullableAnnotationReducer(),
-            new CSharpCastReducer(),
```

Comment:
matches comment above.  also ensures we do less work forking when doing cast reduction work.